feat(docs): add Logto IdP authentication guide (#29037)
expo/expo: docs/pages/guides/authentication.mdx

</Box>
{/* End Keycloak */}

{/* Start Logto */}

<Box
  name="Logto"
  createUrl="https://docs.logto.io/quick-starts/expo/"
  image={ASSETS.logto}
>

| Website                                                     | Provider | PKCE      | Auto Discovery |
| ----------------------------------------------------------- | -------- | --------- | -------------- |
| [Get Your Config](https://docs.logto.io/quick-starts/expo/) | OpenID   | Supported | Available      |

- For native platforms, a Private-Use native URI scheme is required for `redirectUri`. See [OAuth2 spec](https://datatracker.ietf.org/doc/html/rfc8252#section-8.4) for more details.
- For web platform, a `http(s)://` scheme is required for `redirectUri`.

{/* prettier-ignore */}

> ```jsx Auth code  

Reviewer: ````suggestion
```jsx Auth code
````

</Box>
{/* End Uber */}

{/* Start Logto */}

<Box
  name="Logto"
  createUrl="https://docs.logto.io/quick-starts/expo/"
  image={ASSETS.logto}
>

| Website                    | Provider | PKCE      | Auto Discovery |
| -------------------------- | -------- | --------- | -------------- |

> | [Get Your Config][c-logto] | OpenID   | Supported | Available      |

Reviewer: ```suggestion
| [Get Your Config](https://docs.logto.io/quick-starts/expo/) | OpenID   | Supported | Available      |
```

Let's add the link here since it is only used once in this section.

> [c-logto]: https://docs.logto.io/quick-starts/expo/

Reviewer: ```suggestion
```

Let's delete this, see previous comment.

- For native platforms, a Private-Use native URI scheme is required for `redirectUri`. See [OAuth2 spec](https://datatracker.ietf.org/doc/html/rfc8252#section-8.4) for more details.

> - For web platforms,an `http(s)://` scheme is required for `redirectUri`.

Reviewer: ```suggestion
- For web platform, a `http(s)://` scheme is required for `redirectUri`.
```

Let's fix the typos here.

> <Tabs tabs={["Auth Code"]}>
> <Tab>

Reviewer: ```suggestion
```

No need for tabs in this example.

{/* prettier-ignore */}

> ```tsx

Reviewer: ````suggestion
```jsx Auth code
````

Can just add the snippet title here. Also, looking at the code snippet below, there's no need for TypeScript code.

import { LogtoProvider, useLogto } from "@logto/rn";

// Use `useLogto()` hook to sign in and sign out
const Content = () => {
  const { signIn, signOut, isAuthenticated } = useLogto();

  return isAuthenticated ? (
    <Button title="Sign Out" onPress={signOut} />
  ) : (
    <Button title="Sign In" onPress={async () => signIn(redirectUri)} />
  );
};

// Wrap your page component with `LogtoProvider`
const App = () => {
  const logtoConfig = {
    appId: "YOUR_APP",
    endpoint: "YOUR_LOGTO_ENDPOINT",
  };

  return (
    <LogtoProvider config={logtoConfig}>
      <Content />
    </LogtoProvider>
  );
};
```

> </Tab>
> </Tabs>

Reviewer: ```suggestion
```

Let's remove it, see previous comment.
